### PR TITLE
expand FragmentParameter with new properties

### PR DESCRIPTION
**Pull request by wrzontek**
## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Code formatting checked
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

**Comment by wrzontek** [comment]
a lot of the files changed are just import changes from moving `FixedExpressionValue` to common-api

File: interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/FragmentComponentDefinitionExtractor.scala [hunk 1]
```diff
@@ -76,25 +76,34 @@ class FragmentComponentDefinitionExtractor(componentConfig: String => Option[Sin
...
         .value[List[FragmentParamClassLoadErrorData], TypingResult](Unknown)
-        .tell(List(FragmentParamClassLoadErrorData(paramName, p.typ.refClazzName)))
-      ).map(toParameter(componentConfig, paramName, _))
+        .tell(List(FragmentParamClassLoadErrorData(fragmentParameter.name, fragmentParameter.typ.refClazzName)))
+      ).map(toParameter(componentConfig, _, fragmentParameter))
   }
 
-  private def toParameter(componentConfig: SingleComponentConfig, paramName: String, typ: typing.TypingResult) = {
-    val config = componentConfig.params.flatMap(_.get(paramName)).getOrElse(ParameterConfig.empty)
+  private def toParameter(componentConfig: SingleComponentConfig, typ: typing.TypingResult, fragmentParameter: FragmentParameter) = {
```
Comment (arkadius) [inline interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/FragmentComponentDefinitionExtractor.scala:88]: What about validations that the fixedValues and the defaultValue have the same type as the declared type?

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/integration/BaseFlowTest.scala [hunk 2]
```diff
@@ -46,7 +46,7 @@ class BaseFlowTest
     with WithTestHttpClient
     with Matchers
     with OptionValues
-    with EitherValuesDetailedMessage {
+    with EitherValuesDetailedMessage { // todo test fixed value presets here
```
[the remark this answers is not in the document]
Comment (wrzontek) [reply designer/server/src/test/scala/pl/touk/nussknacker/ui/integration/BaseFlowTest.scala:49]: this

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/fixedvaluespresets/FixedValuesPresetProviderFactory.scala [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package pl.touk.nussknacker.engine.api.fixedvaluespresets
+
+import com.typesafe.config.Config
+import sttp.client3.SttpBackend
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait FixedValuesPresetProviderFactory {
+
+  def create(config: Config, sttpBackend: SttpBackend[Future, Any])(
```
[the remark this answers is not in the document]
Comment (wrzontek) [reply components-api/src/main/scala/pl/touk/nussknacker/engine/api/fixedvaluespresets/FixedValuesPresetProviderFactory.scala:10]: it's so the implementation can use it, the Default doesn't, but  it's useful to have the option, as with other providers (for example `AdditionalUIConfigProviderFactory`)

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/fixedvaluespresets/DefaultFixedValuesPresetProvider.scala [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package pl.touk.nussknacker.engine.api.fixedvaluespresets
+
+import pl.touk.nussknacker.engine.api.definition.FixedExpressionValue
+
+class DefaultFixedValuesPresetProvider(
```
[the remark this answers is not in the document]
Comment (wrzontek) [reply components-api/src/main/scala/pl/touk/nussknacker/engine/api/fixedvaluespresets/DefaultFixedValuesPresetProvider.scala:5]: "Default" is consistent with other API classes that allow user implementations

File: scenario-api/src/main/scala/pl/touk/nussknacker/engine/graph/node.scala [hunk 5]
```diff
@@ -298,7 +298,18 @@ object node {
...
-    @JsonCodec case class FragmentParameter(name: String, typ: FragmentClazzRef)
+    @JsonCodec case class FixedExpressionValue(expression: String, label: String)
+
+    @JsonCodec case class FragmentParameter(
+        name: String,
+        typ: FragmentClazzRef,
+        required: Boolean = false,
+        fixedValueList: List[FixedExpressionValue] = List.empty,
+        fixedValueListPresetId: Option[String] = None,
+        allowOnlyValuesFromFixedValuesList: Boolean = false,
```
[the remark this answers is not in the document]
Comment (wrzontek) [reply scenario-api/src/main/scala/pl/touk/nussknacker/engine/graph/node.scala]: i've split it into subclasses to make it more rigid

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/fixedvaluespresets/DefaultFixedValuesPresetProviderFactory.scala [hunk 6]
```diff
@@ -0,0 +1,26 @@
+package pl.touk.nussknacker.engine.api.fixedvaluespresets
+
+import com.typesafe.config.Config
+import pl.touk.nussknacker.engine.api.definition.FixedExpressionValue
+import sttp.client3.SttpBackend
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DefaultFixedValuesPresetProviderFactory extends FixedValuesPresetProviderFactory {
```
[the remark this answers is not in the document]
Comment (wrzontek) [reply components-api/src/main/scala/pl/touk/nussknacker/engine/api/fixedvaluespresets/DefaultFixedValuesPresetProviderFactory.scala:9]: again, I think Default is just more consistent with current naming

**Comment by wrzontek** [comment]
I don't like how this PR evolved, I'll remodel it

File: scenario-api/src/main/scala/pl/touk/nussknacker/engine/graph/node.scala [hunk 7]
```diff
@@ -298,7 +303,78 @@ object node {
...
+      sealed trait FragmentParameterInputMode
+
+      case object InputModeAny extends FragmentParameterInputMode
+
+      @JsonCodec
+      sealed trait InputModeWithFixedList extends FragmentParameterInputMode
+
+      case object InputModeAnyWithSuggestions extends InputModeWithFixedList
+
+      case object InputModeFixedList extends InputModeWithFixedList
```
Comment (bartektartanus) [inline scenario-api/src/main/scala/pl/touk/nussknacker/engine/graph/node.scala:320]: `InputModeFixedList` and `InputModeWithFixedList` are almost identical to me. Is this "middle" trait really needed? Looks like the domain model is more complex to save us one `case` in `match`?

File: scenario-api/src/main/scala/pl/touk/nussknacker/engine/graph/node.scala [hunk 8]
```diff
@@ -298,7 +303,78 @@ object node {
...
+
+    case class FragmentParameterFixedListPreset(
+        name: String,
+        typ: FragmentClazzRef,
+        required: Boolean = false,
+        initialValue: Option[FixedExpressionValue],
+        hintText: Option[String],
+        inputMode: InputModeWithFixedList,
+        fixedValuesListPresetId: String,
+        effectiveFixedValuesList: List[
```
Comment (bartektartanus) [inline scenario-api/src/main/scala/pl/touk/nussknacker/engine/graph/node.scala:374]: What does `effective` mean? Is this the last known value of preset?

**Comment by wrzontek** [comment]
i'll split it into two smaller PR, the first one will be without the preset functionality

**Pull request closed without merging**